sycl/include/sycl/ext/oneapi/latency_control/properties.hpp

```diff
@@ -0,0 +1,86 @@
+//==--- properties.hpp - SYCL properties associated with latency_control ---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/ext/oneapi/properties/property.hpp>
+#include <sycl/ext/oneapi/properties/property_value.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+enum class latency_control_type {
+  none, // default
+  exact,
+  max,
+  min
+};
+
+struct latency_anchor_id_key {
+  template <int Anchor>
+  using value_t = property_value<latency_anchor_id_key,
+                                 std::integral_constant<int, Anchor>>;
+};
+
+struct latency_constraint_key {
+  template <int Target, latency_control_type Type, int Cycle>
+  using value_t =
+      property_value<latency_constraint_key,
+                     std::integral_constant<int, Target>,
+                     std::integral_constant<latency_control_type, Type>,
+                     std::integral_constant<int, Cycle>>;
+};
+
+template <int Target, latency_control_type Type, int Cycle>
+struct property_value<latency_constraint_key,
+                      std::integral_constant<int, Target>,
+                      std::integral_constant<latency_control_type, Type>,
+                      std::integral_constant<int, Cycle>> {
+  using key_t = latency_constraint_key;
+  static constexpr int target = Target;
+  static constexpr latency_control_type type = Type;
+  static constexpr int cycle = Cycle;
+};
+
+template <int Anchor>
+inline constexpr latency_anchor_id_key::value_t<Anchor> latency_anchor_id;
+template <int Target, latency_control_type Type, int Cycle>
+inline constexpr latency_constraint_key::value_t<Target, Type, Cycle>
+    latency_constraint;
+
+template <> struct is_property_key<latency_anchor_id_key> : std::true_type {};
+template <> struct is_property_key<latency_constraint_key> : std::true_type {};
+
+template <typename syclObjectT>
+struct is_property_key_of<latency_anchor_id_key, syclObjectT> : std::true_type {
+};
+template <typename syclObjectT>
+struct is_property_key_of<latency_constraint_key, syclObjectT>
+    : std::true_type {};
```

Comment:
Are these properties intended to be applicable to a class?

- If yes, this should be specialized for the properties and that class specifically.
- If no, this should not be specialized for the properties at all.

Reply:
No, these properties should only apply to functions.
I'll remove the specializations of `is_property_key_of` then.